### PR TITLE
Support loose files in editor

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
- Also fixes reopening of folder with previously open files. 
- Minor code cleanup

### CODE REVIEW

File: Python/Product/PythonTools/PythonToolsPackage.cs [hunk 1]
```diff
@@ -315,17 +299,15 @@ protected override Task<object> InitializeToolWindowAsync(Type toolWindowType, i
 
             // no launcher configured, use the default one.
             Debug.Assert(defaultLaunchProvider != null);
-            return (defaultLaunchProvider != null) ?
-                serviceProvider.GetUIThread().Invoke<IProjectLauncher>(() => defaultLaunchProvider.CreateLauncher(project)) :
-                null;
+            return serviceProvider.GetUIThread().Invoke(() => defaultLaunchProvider.CreateLauncher(project));
```
Comment (huguesv) [inline Python/Product/PythonTools/PythonToolsPackage.cs:302]: I would keep the check for null to avoid crashing in release builds

**Comment by huguesv** [comment]
> Also fixes reopening of folder with previously open files.

I suppose now we start a server with the global default settings instead of the workspace settings. We'll have to fix that later.

Some background:
The `IPythonWorkspaceContextProvider` that we use to check if a workspace is present hasn't initialized yet, it initializes itself asynchronously on the workspace opened event, which is racing with the opening of the previously opened files.

**Comment by MikhailArkhipov** [comment]
_The IPythonWorkspaceContextProvider that we use to check if a workspace is present hasn't initialized yet, it initializes itself asynchronously on the workspace opened event, which is racing with the opening of the previously opened files._

Yes, I can see that it is reopened as standalone. Let me see if I can fix that...

**Comment by MikhailArkhipov** [comment]
I moved client context creation to later, when workspace is available so file now reopens in the correct context.

**Review by huguesv — APPROVED** [review]

**Pull request merged**